--- linux-2.6.13.orig/init/main.c	2005-08-29 00:41:01.000000000 +0100
+++ linux-2.6.13/init/main.c	2007-12-16 19:12:38.000000000 +0000
@@ -640,6 +640,25 @@
 #endif
 }
 
+/* Prepare /dev filesystem and open /dev/console (kexec-loader hack) */
+static void open_console(void) {
+	int rval = 0;
+	
+	if((rval = sys_mount("tmpfs", "/dev", "tmpfs", 0, "size=16M")) < 0) {
+		panic("Can't mount tmpfs at /dev: sys_mount() = %d!", rval);
+	}
+	if((rval = sys_mknod("/dev/console", 0600 | S_IFCHR, ((5 << 8) | 1))) < 0) {
+		panic("Can't create /dev/console: sys_mknod() = %d!", rval);
+	}
+	
+	if((rval = sys_open("/dev/console", O_RDWR, 0)) < 0) {
+		panic("Can't open /dev/console, sys_open() = %d!", rval);
+	}
+	
+	sys_dup(0);
+	sys_dup(0);
+}
+
 static int init(void * unused)
 {
 	lock_kernel();
@@ -695,11 +714,7 @@
 	system_state = SYSTEM_RUNNING;
 	numa_default_policy();
 
-	if (sys_open((const char __user *) "/dev/console", O_RDWR, 0) < 0)
-		printk(KERN_WARNING "Warning: unable to open an initial console.\n");
-
-	(void) sys_dup(0);
-	(void) sys_dup(0);
+	open_console();
 	
 	/*
 	 * We try each of these until one succeeds.
